### ISSUE TITLE
Feature Request: (Optional) Geolocation for faster tagging, more relevant results, support for internal barcode ranges

### ISSUE THREAD

**Issue by bredowmax** [openfoodfacts/openfoodfacts-ios#566]
**Summary**
Allow the iOS app to collect geolocation of the user when scanning a product. This may make tagging more convenient (e.g. serve suggestions for Hunger Games for country, city, retailer), and it may result in more relevant search results when scanning ambiguous barcodes.

**Problem**
Being in US, I came across quite a few product scans within the RCN-range that resulted in completely different product results from barcode scans. I feel this could be avoided if only the app knew my location (In US? At Trader Joe's? Then it's probably a product being sold in US, and not a French product). 
For me, a language on the phone (I'm German but my phone is set to English), the settings in an app (when using El Coco, it thinks I am in Spain) or the language of the product (I like authentic Italian food, but that doesn't mean I'm in Italy) may be indicators of where I am, but if we have access to geolocation - why not use it.

**Open Questions**
Where is a user scanning - at home or in the store? 
We may want to limit location to a country only or map it with a list of supermarkets, to suggest results from this particular shop, or suggest tagging this product with "Available in Country" and "Retailer" if that particular retailer is within 100m of the location of where the photo was taken

**Privacy Concerns**
TLDR: Make it optional.
I understand some users may be concerned about privacy and that is a valid concern. Personally, I wouldn't care. To me, it's not any more intrusive than Google Maps, which I already use. As long as this feature is optional, I would definitely use it.

PS: We could also extract that same information from .Exif files when user takes a photo. This may feel less intrusive

**Comment by teolemon** [comment]
This summary would be a good candidate to be moved on the wiki. The bug tracker is for more granular issues that can be solved one, after the other.
